src/class/eventManagement/dispatcher/events.class.ts

```diff
@@ -0,0 +1,269 @@
+
+    for (const [eventName, validationSchema] of Object.entries(eventsSchema)) {
+      this.#eventsValidationFn.set(eventName, ajv.compile(validationSchema));
+    }
+  }
+
+  public async dispatch(options: DispatchEventOptions<T>): Promise<void> {
+    const { channel, store, redisMetadata, event } = options;
+
+    const transaction = await store.setTransaction({
+      ...event,
+      redisMetadata: {
+        ...event.redisMetadata,
+        ...redisMetadata
+      } as any
+    });
+
+    await channel.publish({
+      ...event,
+      redisMetadata: {
+        ...event.redisMetadata,
+        eventTransactionId: redisMetadata.eventTransactionId,
+        transactionId: transaction.redisMetadata.transactionId
+      }
+    });
+  }
+
+  public async handleEvents(
+    channel: string,
+    event: AnyDispatcherChannelMessage |
+    AnyIncomerChannelMessage<T>
+  ) {
+    if (!event.name || !event.redisMetadata) {
+      throw new Error("Malformed message");
+    }
+
+    if (event.redisMetadata.origin === this.privateUUID) {
+      return;
+    }
+
+    if (channel === this.dispatcherChannelName) {
+      if (!isDispatcherChannelMessage(event)) {
+        throw new Error("Unknown event on Dispatcher Channel");
+      }
+
+      try {
+        this.dispatcherChannelMessagesSchemaValidation(event);
+      }
+      catch (error) {
+        this.#logger.error(this.#standardLogFn({
+          ...event,
+          redisMetadata: {
+            ...event.redisMetadata,
+            origin: this.privateUUID
+          }
+        } as unknown as StandardLogOpts<T>)(error.stack));
+      }
+
+      match<DispatcherChannelEvents>({ name: event.name })
+        .with({ name: "REGISTER" }, async() => {
+          this.#logger.info(this.#standardLogFn(event as any)("Registration asked"));
+          // this.#logger.info({ channel, event }, "Registration asked");
+
+          this.emit("APPROVEMENT", event);
+        })
+        .otherwise(() => {
+          throw new Error("Unknown event on Dispatcher Channel");
+        });
+    }
+    else if (isIncomerChannelMessage(event)) {
+      this.incomerChannelMessagesSchemaValidation(event);
+
+      if (isCloseMessage(event)) {
+        this.emit("CLOSE", channel, event);
+
+        return;
+      }
+
+      this.emit("CUSTOM_EVENT", channel, event);
+    }
+    else {
+      throw new Error("Unknown event for the given Channel");
+    }
+  }
+
+  private redisMetadataValidation(event: AnyDispatcherChannelMessage | AnyIncomerChannelMessage<T>) {
+    const { redisMetadata } = event;
+
+    const redisMetadataValidationFn = this.#eventsValidationFn.get("redisMetadata") as ValidateFunction<T>;
+
+    if (!redisMetadataValidationFn(redisMetadata)) {
+      throw new Error(
+        `Malformed redis metadata: [${[...redisMetadataValidationFn.errors]
+          .map((error) => `${error.instancePath ? `${error.instancePath}:` : ""} ${error.message}`).join("|")}]`
```

Comment:
I see the same thing 3 times. It would be possible to make a method ? For example : 
```ts
function concatErrors(errors: ErrorType[]) {
  return errors.map((error) => ...)
}
```